### PR TITLE
Adding TLS mutual auth supoort to jti_openconfig_telemetry plugin

### PR DESCRIPTION
**Pull request by littlespace**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go [hunk 1]
```diff
@@ -74,10 +80,14 @@ var (
...
-  ssl_cert = "/etc/telegraf/cert.pem"
-
+  ## enable client-side TLS and define CA to authenticate the device
+  # enable_tls = true
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # insecure_skip_verify = true
+  ## define client-side TLS certificate & key to authenticate to the device
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+
```
Comment (danielnelson) [inline plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go]: Let's use the typical comment block here for consistency with other plugins:
```
  ## Optional TLS Config
  # tls_ca = "/etc/telegraf/ca.pem"
  # tls_cert = "/etc/telegraf/cert.pem"
  # tls_key = "/etc/telegraf/key.pem"
  ## Use TLS but skip chain & host verification
  # insecure_skip_verify = false
```

File: plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go [hunk 2]
```diff
@@ -374,8 +383,8 @@ func (m *OpenConfigTelemetry) Start(acc telegraf.Accumulator) error {
 		}
 
 		// If a certificate is provided, open a secure channel. Else open insecure one
-		if transportCredentials != nil {
-			grpcClientConn, err = grpc.Dial(server, grpc.WithTransportCredentials(transportCredentials))
+		if tlscfg != nil {
+			grpcClientConn, err = grpc.Dial(server, grpc.WithTransportCredentials(credentials.NewTLS(tlscfg)))
```
Comment (danielnelson) [inline plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go]: All the code is right, but if you are interested in going further I can see a way we could have done this nicer originally. We should have created a `grpc.DialOption` above based on EnableTLS, and then here we don't need the if/else.  Optional.

File: plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go (hunk 1 above)
Comment (littlespace) [reply plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go]: done

File: plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go (hunk 2 above)
Comment (littlespace) [reply plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go]: done

**Pull request merged**